src/Parallel/CharmPupable.hpp

```diff
@@ -16,8 +16,8 @@
  * \ingroup ParallelGroup
  * \brief Mark derived classes as serializable
  *
- * Any class that derives off of a non-class template base class must contain
- * this macro if it is to be serialized.
+ * Any class that derives off of a base class that is not a template must
```

Comment:
[a little pedantic, but this is a rewording PR so :shrug:]
I suggest `derives off of` -> `inherits from` and refer to the potential base class as a `template instantiation` or `not a template instantiation` (because ofc you can't inherit from a template only the instantiation thereof).
Presumably also this is specific to only abstract base classes? If my understanding is correct then also suggest `base`->`abstract base`

Reply:
It's not an instantiation though. It is a specialization, or, more technically, a template class (which ones the previous wording that was confusing). I've updated the wording using `template class` but in a way that is hopefully not confusing.